### PR TITLE
Various changes related to hearts

### PR DESCRIPTION
**Pull request by tobiasvl**
- Split recalculation of maximum hearts out from `__PATCH_4__` into `__RECALCULATE_MAX_HEARTS__`
- Document routines that clamp/recalculate maximum hearts
- Rename everything related to "MaxHealth" to "MaxHearts" for consistency
- Document the routine that draws hearts in the file menu
- Misc

### CODE REVIEW

File: src/code/bank1.asm [hunk 1]
```diff
@@ -1257,102 +1257,172 @@ jr_001_5C7B::
...
-    db  $00                                       ; $5D3D
-
-Data_001_5D3E::
-    db  $99, $8B, $06, $7E, $7E, $7E, $7E, $7E, $7E, $7E ; $5D3E
-    db  $99, $AB, $06, $7E, $7E, $7E, $7E, $7E, $7E, $7E ; $5D48
-    db  $00                                       ; $5D52
+SaveSlot1HeartsDrawData::
+    ; First row:
+    db  $98 ; wDrawCommand.destinationHigh
+    db  $CB ; wDrawCommand.destinationLow
```
Comment (kemenaran) [inline src/code/bank1.asm:1263]: Nitpick, and really not important: maybe 

```suggestion
    dw  $98CB ; wDrawCommand.destination
```

?

File: src/code/bank1.asm [hunk 2]
```diff
@@ -1257,102 +1257,172 @@ jr_001_5C7B::
...
-label_001_5D53::
+; Builds a wDrawCommand to draw hearts next to the save files.
+;
+; Draws one full heart per full heart remaining in the actual health,
+; then one full heart if there's half a heart remaining of the actual health,
+; then one full heart per empty heart of the max health.
+;
+; This seems to have originally supported drawing the actual health of the
+; player at the time of saving, but the final code (in all revisions of both
+; the original game and DX) simply draws full hearts for all Heart Containers.
```
Comment (kemenaran) [inline src/code/bank1.asm:1310]: Ha, neat, I didn't knew that!

File: src/code/entities/bank3.asm [hunk 3]
```diff
@@ -2380,8 +2380,8 @@ SpawnEnemyDrop::
     ld   a, [wGuardianAcornCounter]               ; $55EE: $FA $71 $D4
     inc  a                                        ; $55F1: $3C
     ld   [wGuardianAcornCounter], a               ; $55F2: $EA $71 $D4
-    cp   GUARDIAN_ACCORN_COUNTER_MAX              ; $55F5: $FE $0C
-    jr   c, .noGuardianAccornDrop                 ; $55F7: $38 $16
+    cp   GUARDIAN_ACORN_COUNTER_MAX               ; $55F5: $FE $0C
```
Comment (kemenaran) [inline src/code/entities/bank3.asm:2383]: Typos, typos, typos… 😅

File: src/code/photos.asm [hunk 4]
```diff
@@ -3130,7 +3131,7 @@ JumpTable_037_5393.func_037_53C0::
     ld   [wBGMapToLoad], a                        ; $53FA: $EA $FF $D6
     ret                                           ; $53FD: $C9
 
-func_037_53FE::
+IncrementGameplaySubtype_37::
```
Comment (kemenaran) [inline src/code/photos.asm:3134]: Yay for labeled utility functions

File: src/constants/memory/wram.asm [hunk 5]
```diff
@@ -3556,28 +3567,28 @@ wFile3DeathCountHigh::
...
+; Amount of health for file 2, needs to be wFile2MaxHearts * 8 or the game crashes while drawing the file selection menu
 wFile2Health::
   ds 1 ; DC07
 
-; Amount of health for file 3, needs to be wFile3MaxHealth * 8 or the game crashes while drawing the file selection menu
+; Amount of health for file 3, needs to be wFile3MaxHearts * 8 or the game crashes while drawing the file selection menu
 wFile3Health::
   ds 1 ; DC08
 
 ; Amount of maximum health for file 1, used to draw the hearts on the file selection menu
```
Comment (kemenaran) [inline src/constants/memory/wram.asm:3582]:
```suggestion
; Amount of maximum hearts for file 1, used to draw the hearts on the file selection menu
```

?

**Review by kemenaran — APPROVED** [review]

File: src/code/bank1.asm (hunk 1 above)
Comment (tobiasvl) [reply src/code/bank1.asm:1263]: Those comments refer to the corresponding labels here, and there is no `destination` label: https://github.com/zladx/LADX-Disassembly/blob/main/src/constants/memory/wram.asm#L2876

Also, the Game Boy is little-endian, even though this struct is big-endian for some reason, so calling it `destination` would perhaps also confuse people who expect to be able to use the whole thing as an address? :)

File: src/constants/memory/wram.asm (hunk 5 above)
Comment (tobiasvl) [reply src/constants/memory/wram.asm:3582]: Quite right, I'll fix it for all three file comments and rebase

**Pull request merged**